File: gcc/testsuite/c-c++-common/gomp/target-device-ancestor-4.c
/* { dg-do compile } */
/* { dg-additional-options "-fdump-tree-original" } */

  /* Test to ensure that device-modifier 'ancestor' is parsed correctly in
     device clauses. */

#pragma omp requires reverse_offload /* { dg-message "sorry, unimplemented: 'reverse_offload' clause on 'requires' directive not supported yet" } */

void
foo (void)
{
  #pragma omp target device (ancestor: 1) /* { dg-message "" "sorry, unimplemented: 'ancestor' not yet supported" { xfail *-*-* } } */
  ;

}

/* { dg-final { scan-tree-dump "pragma omp target \[^\n\r)]*device\\(ancestor:1\\)" "original" } } */

File: gcc/testsuite/gfortran.dg/gomp/target-device-ancestor-4.f90
! { dg-do compile }
! { dg-additional-options "-fdump-tree-original" }

! Test to ensure that device-modifier 'ancestor' is parsed correctly in
! device clauses.

!$omp requires reverse_offload

!$omp target device (ancestor : 1)
!$omp end target

end

! { dg-final { scan-tree-dump-times "pragma omp target \[^\n\r)]*device\\(ancestor:1\\)" 1 "original" } }
